engine/src/player.ts

```diff
@@ -161,12 +161,18 @@ export default class Player extends EventEmitter {
     return json;
   }
 
-  static fromData(data: any, map: SpaceMap, customization: FactionCustomization, expansions: number) {
+  static fromData(data: any, map: SpaceMap, customization: FactionCustomization, expansions: number, phase: Phase) {
     const player = new Player(data.player);
-    player.factionVariant = data.factionVariant;
 
-    if (data.faction) {
-      player.loadFaction(data.faction, customization, expansions, true);
+    const isFactionSelected =
+      phase !== Phase.SetupInit &&
+      phase !== Phase.SetupBoard &&
+      phase !== Phase.SetupFaction &&
+      phase !== Phase.SetupAuction;
```

Comment:
I think you can just test `if (data.board)` to know if a faction has been loaded

Reply:
Actually it's not exported in the json - my bad.

Exporting `factionLoaded` boolean based on `!!data.board` would be a solution - but it breaks compatibility -- unless we pass the engine version to `fromData`.

I still prefer this latter solution - it's less susceptible to regressions, the `Player` doesn't have to know the flow of the engine, and adding an additional phase at setup won't cause issues.

Reply:
Alright, sounds good! I'm all in favor of trying to separate those concerns. I will look at that tomorrow.

Reply:
Hmz. Apparently just moving the line `this.factionVariant = data.factionVariant` into the `if` statement fixes it. It will unload the faction variant when saving between bids. It feels really hacky to just leave it at that though, as the engine/player then depends on how it is used. So I still went with the suggestion of the `factionLoaded`.